# HG changeset patch
# Parent 55fad75d03418ec617736ab189a9a542a440d7ea
Doing a full global TLB flush every 1024 regular TLB flushes causes some
performance issues.  Remove the 1024 limit even in the case of a debug build.

diff -r 55fad75d0341 xen/arch/x86/flushtlb.c
--- a/xen/arch/x86/flushtlb.c
+++ b/xen/arch/x86/flushtlb.c
@@ -13,12 +13,7 @@
 #include <asm/flushtlb.h>
 #include <asm/page.h>
 
-/* Debug builds: Wrap frequently to stress-test the wrap logic. */
-#ifdef NDEBUG
 #define WRAP_MASK (0xFFFFFFFFU)
-#else
-#define WRAP_MASK (0x000003FFU)
-#endif
 
 u32 tlbflush_clock = 1U;
 DEFINE_PER_CPU(u32, tlbflush_time);
